src/main/java/com/cloudbees/jenkins/GitHubRepositoryNameContributor.java

```diff
@@ -90,20 +96,19 @@ public void parseAssociatedNames(AbstractProject<?, ?> job, Collection<GitHubRep
      * MultiSCM support separated into a different extension point since this is an optional dependency
      */
     @Extension(optional=true)
-    @SuppressWarnings("unused")
     public static class FromMultiSCM extends AbstractFromSCMImpl {
         // make this class fail to load if MultiSCM is not present
         public FromMultiSCM() { MultiSCM.class.toString(); }
 
         @Override
-        public void parseAssociatedNames(AbstractProject<?, ?> job, Collection<GitHubRepositoryName> result) {
-            if (job.getScm() instanceof MultiSCM) {
-                EnvVars env = buildEnv(job);
-
-                MultiSCM multiSCM = (MultiSCM) job.getScm();
-                List<SCM> scmList = multiSCM.getConfiguredSCMs();
-                for (SCM scm : scmList) {
-                    addRepositories(scm, env, result);
+        public void parseAssociatedNames(Job<?, ?> job, Collection<GitHubRepositoryName> result) {
+            SCMTriggerItem item = SCMTriggerItems.asSCMTriggerItem(job);
+            EnvVars env = buildEnv(job);
+            for (SCM scm : item.getSCMs()) {
+                if (scm instanceof MultiSCM){
+                    MultiSCM multiSCM = (MultiSCM) scm;
+                    List<SCM> scmList = multiSCM.getConfiguredSCMs();
+                    addRepositories(scmList, env, result);
```

Comment:
`FromMultiSCM` can probably be deleted entirely (and the dependency on the `multiple-scms` plugin removed!) since [`Project.getSCMs` already handles this case](https://github.com/jenkinsci/jenkins/blob/0016f85a500ca74a90358177c3cdb51ad7c8e5d2/core/src/main/java/hudson/model/Project.java#L120-122).